### ISSUE TITLE
Docker remote attr remote_repo_layout_ref repeatedly re-applies when set

### ISSUE THREAD

**Issue by diginc** [jfrog/terraform-provider-artifactory#542]
**Describe the bug**

When setting the common argument [remote_repo_layout_ref](https://registry.terraform.io/providers/jfrog/artifactory/latest/docs/resources/remote#remote_repo_layout_ref) on [remote_docker_repository](https://registry.terraform.io/providers/jfrog/artifactory/latest/docs/resources/remote_docker_repository) it never seems to actually get set.  

## Code Example

```
resource "artifactory_remote_docker_repository" "my-remote-docker" {
  key                            = "my-remote-docker"
  external_dependencies_enabled  = true
  external_dependencies_patterns = ["**/hub.docker.io/**"]
  enable_token_authentication    = true
  url                            = "https://hub.docker.io/"
  block_pushing_schema1          = true
  # Added this to documentation's example
  remote_repo_layout_ref         = "simple-default"  
}


# First apply will show this attribute getting set along with all the others
# Every 2nd+ apply results in this diff

Terraform will perform the following actions:

  # artifactory_remote_docker_repository.my-remote-docker will be updated in-place
  ~ resource "artifactory_remote_docker_repository" "my-remote-docker" {
        id                                    = "my-remote-docker"
      + remote_repo_layout_ref                = "simple-default"
        # (32 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

```

## Version

```
# Artifactory
  "version" : "7.41.6",
  "revision" : "74106900"

# Terraform Provider
provider "registry.terraform.io/jfrog/artifactory" {
  version     = "6.12.0"

provider "registry.terraform.io/hashicorp/aws" {
  version     = "4.27.0"

# CLI tools

 % terraform version
Terraform v1.2.9
on linux_amd64

 % terragrunt -v
terragrunt version v0.38.12

```

**Requirements for and issue**
- [X] A description of the bug
- [X] A fully functioning terraform snippet that can be copy&pasted (no outside files or ENV vars unless that's part of the issue). **If this is not supplied, this issue will likely be closed without any effort expended.**
- [x] Your version of artifactory (you can `curl` it at `$host/artifactory/api/system/version`
- [x] Your version of terraform
- [x] Your version of terraform provider

**Expected behavior**
After the first apply terraform should say 'no changes' and the setting should be visible in the web UI as configured.

**Additional context**
Web UI does not show the setting as set either.  

For other repo types using this attr, I've only tested conan so far and it seems to not repeatedly apply the setting change.  However the setting still doesn't show up in the webui or config descriptor XML.  I haven't dug too deep into that but I'm adding a bunch more repo types in the coming days so might discover more.

As a workaround we're just not setting `remote_repo_layout_ref ` since docker commands seems to function fine without this set. Thought you might like to know though.

**Comment by alexhung** [comment]
@diginc Thanks for the report. We'll take a look.

**Comment by danielmkn** [comment]
I've looked into the problem. 
[Create remote repository](https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-CreateRepository) call consumes [this body](https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-CreateRepository), which, actually, doesn't have the attribute `remoteRepoLayoutRef`. 
The provider sends the attribute to Artifactory, but never gets it back in the response body because it doesn't exist. `remoteRepoLayoutRef` is simply got ignored. 
Also, the field was set as `Computed` in the schema, which means it is supposed to be set when the provider gets a response from the Artifactory instance. From the TF documentation:
```
	// Computed indicates whether the provider may return its own value for
	// this attribute or not.
```
It looks like a legacy field. 
@alexhung, I suggest removing this attribute from the resource, documentation, and all the examples.

**Comment by danielmkn** [comment]
@diginc, deprecation note was added in 6.16.1

**Comment by danielmkn** [comment]
@diginc, @alexhung I see that `remote_repo_layout_ref` bug is fixed in Artifactory 7.47.0. Once it's released, we need to test and remove the deprecation notice.
